### DISCUSSION TITLE
Is the .well-known/jmap route supposed to return urls with placeholders?

### DISCUSSION THREAD

**Discussion by lukeh990** [stalwartlabs/jmap-server#43] [Q&A] [answered]
I am trying to use the JMAP demo webmail client and it kept giving me an error and I noticed that the result from the `.well-known/jmap` request included several items that were URLs but had placeholders. Since the client seemed to have expected these URLs to be filled in it didn't work. This is an example of what I mean:

```
{
downloadUrl: "https://jmap.lukeh990.dev/jmap/download/{accountId}/{blobId}/{name}?accept={type}"
eventSourceUrl: 'https://jmap.lukeh990.dev/jmap/eventsource/?types={types}&closeafter={closeafter}&ping={ping}"
}
```

**Comment by zvasilev** [comment, marked as answer]
Yes, it is expected that the .well-known/jmap route have to return a URL with placeholders, as this is one of its intended purposes.
The placeholders allow clients to substitute the appropriate values when making requests to the JMAP endpoint.

**Reply by lukeh990** [reply]
Well thank you for clearing that up. Are you familiar with the JMAP demo client listed on jmap.io? If not can you suggest a client preferably web based that can interact with JMAP?

**Reply by zvasilev** [reply]
I assume you are referring to this client https://jmap.io/jmap-demo-webmail/ ?
The last time I tried it, it was working with fastmail (FM), but that was some time ago. I used it to see how to construct some queries. I am the developer behind mailtemi.com .
Since then, FM changed how you can log in through JMAP. Instead of using an app-password, now OAuth or a token is required.
I have never tried it with stalwart.

**Reply by lukeh990** [reply]
Oh, I have tried using the mailtemi client. I kept getting a "Incorrect Settings For JMAP Service" error then "code(55)". Can you help? The little pop up on the app doesn't explain it well.

**Reply by zvasilev** [reply]
Hi, the error number along with some others doesn't have text messages (for JMAP) yet. 
It meaning is that an account was not found. 
This could be due to not creating an account through Stalwart CLI, using the command 'stalwart-cli -u https://jmap.example.org -c PASSWORD account list' ?
Although I am currently on vacation, I will be more than happy to assist you via email at support@mailtemi.com.
